Airbrake Backend Plugin

moltenice opened the pull request:
## Hey, I just made a Pull Request!

This creates a backend plugin to be used by Airbrake Frontend.

It simply creates a proxy with the API but passing in the API key via a URL parameter as that's how Airbrake handles API Keys being passed in. More details here: https://airbrake.io/docs/api/#authentication

It also follows 100% test coverage just like the frontend plugin for everything except for the code used to start up the standalone instance.

#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

benjdlambert commented on plugins/airbrake-backend/package.json:
```diff
@@ -0,0 +1,57 @@
+    "@backstage/cli": "^0.13.1-next.0",
+    "@types/http-proxy-middleware": "^0.19.3",
+    "@types/supertest": "^2.0.8",
+    "supertest": "^4.0.2",
```
Can you bump this to `supertest@^6.1.6` as it's what we've got in the project already. saves downloading two different versions.

benjdlambert reviewed (commented):
Thanks for this, just some things to work through :pray:

moltenice commented:
@benjdlambert All issues should have been addressed and the PR is ready for a re-review. Do let me know if you spot anything else!

benjdlambert commented on plugins/airbrake-backend/src/config/ExtractAirbrakeConfig.ts:
```diff
@@ -0,0 +1,41 @@
+ */
+export function extractAirbrakeConfig(config: Config): AirbrakeConfig {
+  return {
+    apiKey: config.getString('airbrake.apiKey'),
```
Probably makes sense to add a config schema to the backend plugin so that we can be sure that this is going to be here at startup otherwise this will crash. Here's a guide: https://backstage.io/docs/conf/defining

moltenice replied:
Thanks for the review, I've done this I think, any chance you could see if I did it correctly @benjdlambert ?

benjdlambert reviewed (commented):
Sorry just one thing I missed before and think we're ready to go :tada:

benjdlambert commented:
@moltenice just in case you missed this: https://github.com/backstage/backstage/pull/9390#discussion_r801616304

benjdlambert commented on plugins/airbrake-backend/src/config/index.ts:
```diff
@@ -0,0 +1,19 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Config } from '../../config';
```
Probably gonna need to move `config.d.ts` inside this `src` root for this not to throw an error at the `api-report` step it think 👍  And then change the `configSchema` to the correct path.

benjdlambert commented on ADOPTERS.md:
```diff
@@ -1,92 +1,92 @@
-| [VMware](https://www.vmware.com)                                   | [@mpriamo](https://github.com/mpriamo), [@krisapplegate](https://github.com/krisapplegate)                                                     | Part of [Tanzu Application Platform](https://docs.vmware.com/en/VMware-Tanzu-Application-Platform/index.html) offering; internal developer portal  |
-| [Ualá](https://www.uala.com.ar/)                                   | [Santiago Bernal](https://github.com/sabernal)                                                   | Initial work being done to centralize documentation for all our microservices and APIs, as well as scaffolding new services and tracking code quality |
-| [IKEA IT AB](https://www.ingka.com)                                       | [@bjornramberg](https://github.com/bjornramberg), [@supriyachitale](https://github.com/supriyachitale) | Supporting engineers at scale with self serve access and connecting the dots of our engineering platform and services, enabling product teams to move faster and go further, and unleashing innovation, reuse and co-cre ...
+| Organization                                                                | Contact                                                                                                                                                                                                      | Description of Use                                                                                               ...
```
Seems this is back again 🙈

moltenice replied:
Why does this keep happening 😭 I have a feeling a commit hook causes this. Sorry about that, will remove it :)

moltenice commented:
> It seems the ADOPTERS.md change has crept back in again... 🙈 if you do `git checkout origin/master -- ADOPTERS.md` from the root, and commit that should be good to go.

@benjdlambert fixed!

Rugvip commented on plugins/airbrake-backend/package.json:
```diff
@@ -0,0 +1,59 @@
+  "main": "src/index.ts",
+  "types": "src/index.ts",
+  "license": "Apache-2.0",
+  "private": true,
```
👀

moltenice replied:
Woops, fixed!

Rugvip reviewed (approved):
Thank you, looks good! 👍 

Just one minor thing to make sure we actually :shipit:  😁

moltenice commented:
> Thank you, looks good! 👍
> 
> Just one minor thing to make sure we actually :shipit: 😁

Thank you for taking the time to review :) I've addressed your comment, good catch!

Rugvip commented on app-config.yaml:
```diff
@@ -464,3 +464,6 @@ apacheAirflow:
   baseUrl: https://your.gocd.instance.com
+
+airbrake:
+  apiKey: ${AIRBRAKE_API_KEY}
```
Ah, realized one more thing, which is that this is required right? As in if `AIRBRAKE_API_KEY` isn't set the backend will refuse to start up, even in development?

Would like that to not be the case, you could for example make it a soft error in development like some other plugins do. Or take the route that for example `azureDevOps.token` does in this config and just use a mock value

moltenice replied:
Not a problem, I'll take a look and get this sorted in another branch.

Rugvip commented:
Will fix in followup, wanna get this shipped 😅
